### PR TITLE
gProfiler 1.2.7

### PR DESCRIPTION
**Pull request by Jongy**
After:
* https://github.com/Granulate/gprofiler/pull/212
* https://github.com/Granulate/gprofiler/pull/213

### CODE REVIEW

**Comment by Jongy** [comment]
@IlayRosenberg I made some refactors, mainly to add the actual information in the log messages

File: gprofiler/profilers/java.py [hunk 1]
```diff
@@ -430,10 +430,12 @@ def __init__(
         self._interval = int((1 / frequency) * 1000_000_000)
         self._buildids = java_async_profiler_buildids
         self._version_check = java_version_check
-        if not java_version_check:
+        if not self._version_check:
             logger.warning("Java version checks are disabled")
         self._mode = java_async_profiler_mode
         self._safemode = java_async_profiler_safemode
+        if self._safemode:
```
Comment (Jongy) [inline gprofiler/profilers/java.py:437]: Whoopsie

File: gprofiler/profilers/java.py [hunk 2]
```diff
@@ -503,37 +505,49 @@ def _run_java_version() -> None:
...
                 )
-                return None
+                return False
         else:
             if self._version_check and process_basename == "java":
-                if not self._is_jvm_type_supported(self._get_java_version(process)):
-                    logger.error(f"Process {process.pid} running unsupported JVM, skipping...")
-                    return None
+                java_version_output = self._get_java_version(process)
+                if self._check_jvm_type_supported(process, java_version_output):
```
Comment (IlayRosenberg) [inline gprofiler/profilers/java.py:541]: if not?

Comment (Jongy) [reply gprofiler/profilers/java.py:541]: lol, I checked with `--java-safemode` which was okay. This explains why the tests fail

**Review by IlayRosenberg — APPROVED** [review]
LGTM

**Pull request merged**